diff --git a/src/targets/gpu/CMakeLists.txt b/src/targets/gpu/CMakeLists.txt
index 6b5a052..c267e78 100644
--- a/src/targets/gpu/CMakeLists.txt
+++ b/src/targets/gpu/CMakeLists.txt
@@ -377,11 +377,6 @@ if(MIGRAPHX_USE_HIPBLASLT)
     target_link_libraries(migraphx_gpu PUBLIC roc::hipblaslt)
 endif()
 
-if(WIN32)
-    # Temporary workaround on rocMLIR not exporting correctly libraries it depends on.
-    target_link_libraries(migraphx_gpu PRIVATE ntdll)
-endif()
-
 target_link_libraries(migraphx_gpu PUBLIC migraphx)
 if(NOT MIGRAPHX_USE_MIOPEN AND NOT MIGRAPHX_USE_ROCBLAS)
     target_link_libraries(migraphx_gpu PUBLIC migraphx_device)