diff --git a/bin/rocprofv2 b/bin/rocprofv2
index 2a158e8..62dbaa9 100644
--- a/bin/rocprofv2
+++ b/bin/rocprofv2
@@ -7,7 +7,8 @@ if [ -n "${ROCP_PRELOAD}" ]; then LD_PRELOAD="${ROCP_PRELOAD}"; fi
 
 CURRENT_DIR="$( dirname -- "$0"; )";
 ROCPROFV2_DIR=$(dirname -- $(realpath ${BASH_SOURCE[0]}));
-ROCM_DIR=$( dirname -- "$ROCPROFV2_DIR"; )
+ROCPROFILER_DIR=$( dirname -- "$ROCPROFV2_DIR"; )
+ROCM_DIR=@rocmtoolkit_merged@
 PLUGIN_LIST=("ctf" "perfetto" "file" "att" "cli" "json")
 RUN_FROM_BUILD=0
 if [[ $ROCPROFV2_DIR == *"/build"* ]]; then
@@ -15,7 +16,7 @@ if [[ $ROCPROFV2_DIR == *"/build"* ]]; then
   ROCM_DIR=$ROCPROFV2_DIR
 fi
 
-export ROCPROFILER_METRICS_PATH=$ROCM_DIR/libexec/rocprofiler/counters/derived_counters.xml
+export ROCPROFILER_METRICS_PATH=$ROCPROFILER_DIR/libexec/rocprofiler/counters/derived_counters.xml
 export LD_LIBRARY_PATH=$ROCM_DIR/lib:$LD_LIBRARY_PATH
 export HSA_TOOLS_LIB=$ROCM_DIR/lib/librocprofiler64.so.2
 
@@ -87,7 +88,7 @@ while [ 1 ]; do
       exit 1
     fi
   elif [[ "$1" == "--list-counters" ]]; then
-    export LD_PRELOAD=$LD_PRELOAD:$ROCM_DIR/lib/rocprofiler/librocprofiler_tool.so
+    export LD_PRELOAD=$LD_PRELOAD:$ROC_DIR/lib/rocprofiler/librocprofiler_tool.so
     eval $ROCM_DIR/libexec/rocprofiler/ctrl
     exit 1
   elif [[ "$1" == "-i" || "$1" == "--input" ]]; then
@@ -228,7 +229,7 @@ while [ 1 ]; do
       if [ $RUN_FROM_BUILD == 1 ]; then
         ATT_PATH=$ROCM_DIR/plugin/att/att/att.py
       else
-        ATT_PATH=$ROCM_DIR/libexec/rocprofiler/att/att.py
+        ATT_PATH=$ROCPROFILER_DIR/libexec/rocprofiler/att/att.py
         export ROCPROFV2_ATT_LIB_PATH=$ROCM_DIR/lib/hsa-amd-aqlprofile/librocprofv2_att.so
       fi
       ATT_ARGV=$3
@@ -314,13 +315,13 @@ if [ -n "$PMC_LINES" ] && [ ! -n "$ATT_ARGV" ]; then
       export OUTPUT_PATH=$FINAL_PATH
     fi
     let COUNTER=COUNTER+1
-    LD_PRELOAD=$LD_PRELOAD "${@}"
+    LD_PRELOAD=$LD_PRELOAD:$ROCPROFILER_DIR/lib/rocprofiler/librocprofiler_tool.so $*
     if [ -n "$OUTPUT_PATH" ]; then
       echo -e "\nThe output path for the following counters: $OUTPUT_PATH"
     fi
   done
 else
- LD_PRELOAD=$LD_PRELOAD "${@}"
+  LD_PRELOAD=$LD_PRELOAD:$ROCPROFILER_DIR/lib/rocprofiler/librocprofiler_tool.so $*
 fi
 
 get_pmc_results_txt_path() {
