diff --git a/src/gemm_v2.cpp b/src/gemm_v2.cpp
index a20328c..89f3f3a 100644
--- a/src/gemm_v2.cpp
+++ b/src/gemm_v2.cpp
@@ -47,9 +47,9 @@
 #include <rocblas/rocblas.h>
 /// rocblas_gemm_ex3 supports F8 datatypes.
 #ifdef _WIN32
-#define USE_ROCBLAS_GEMM_EX3 ((MIOPEN_ROCBLAS_VERSION_FLAT >= 3000000) && ROCBLAS_BETA_FEATURES_API)
+#define USE_ROCBLAS_GEMM_EX3 false
 #else
-#define USE_ROCBLAS_GEMM_EX3 ((MIOPEN_ROCBLAS_VERSION_FLAT >= 2047000) && ROCBLAS_BETA_FEATURES_API)
+#define USE_ROCBLAS_GEMM_EX3 false
 #endif
 #endif
 #include <miopen/perf_field.hpp>
diff --git a/src/solver/mha/mha_common.hpp b/src/solver/mha/mha_common.hpp
index 6c3ebe4..732d7da 100644
--- a/src/solver/mha/mha_common.hpp
+++ b/src/solver/mha/mha_common.hpp
@@ -44,7 +44,7 @@
 #include <rocblas/rocblas.h>
 /// rocblas_gemm_ex3 supports F8 datatypes.
 /// strided_batched_ex3 introduced in rocblas 4.0
-#define USE_ROCBLAS_EX3 ((MIOPEN_ROCBLAS_VERSION_FLAT >= 4000000) && ROCBLAS_BETA_FEATURES_API)
+#define USE_ROCBLAS_EX3 false
 #endif
 #endif
 
